indico/htdocs/js/indico/modules/contributions/common.js

```diff
@@ -236,6 +236,50 @@
             var target = $(evt.target);
             reloadManagementAttachmentInfoColumn(target.data('locator'), target.closest('td'));
         });
+
+
+        $('#subcontribution-list table').sortable({
+            items: '.js-sortable-subcontribution-row',
+            handle: '.ui-sortable-handle',
+            placeholder: 'sortable-placeholder',
+            tolerance: 'pointer',
+            distance: 10,
+            axis: 'y',
+            containment: '#subcontribution-list table',
+            start: function(e, ui){
+                ui.placeholder.height(ui.helper.outerHeight());
+            },
+            update: function(e, ui) {
+                var self = $(this);
+                var oldIndex = ui.item.data('position');
+                var newIndex = ui.item.index() + 1;
+
+                if (oldIndex === newIndex) {
+                    return;
+                }
+
+                var subcontributionIds = self.sortable('toArray');
+                var positions = {};
+
+                ui.item.data('position', newIndex);
+
+                $.map(subcontributionIds, function(subcontributionId) {
+                    var row = self.find('#' + subcontributionId);
+                    var newIndex = row.index() + 1;
+
+                    positions[subcontributionId] = newIndex;
+                    row.find('.subcontribution-position').text(newIndex);
+                });
+
+                $.ajax({
+                    url: ui.item.data('sort-url'),
+                    method: 'POST',
+                    data: {subcontribution_id: subcontributionIds, positions: JSON.stringify(positions)},
```

Comment:
(why not send the whole payload as json? then you could simply use `request.json`)

actually, instead of the dict where you need to convert the keys back to int: why not send a list of subcontributions? you wouldn't need any json for that (`request.form.getlist(..)` on the server side) and could simple iterate over the entries and set the position, similar to how we do it when reordering survey sections:

```python
        sections = {section.id: section for section in self.survey.sections}
        section_ids = map(int, request.form.getlist('section_ids'))
        for position, section_id in enumerate(section_ids, 1):
            sections[section_id].position = position
```